### PR TITLE
refactor: perform eth_sendTransaction via EVM module

### PR DESCRIPTION
**Pull request by meeh0w**
* [CP-9159](https://ava-labs.atlassian.net/browse/CP-9159)
* _+ migrate `eth_sendTransaction` to be performed via EVM Module_

---

* Related PR: https://github.com/ava-labs/vm-modules/pull/160

## Changes
* Starts using the `updateTx` callback provided by VM Modules for updating transaction fee configs / spend limits (see related PR above)
* Removes `eth_sendTransaction` handler and some of the related files so that those requests are performed using the `EvmModule`
    * Also removes the `SignTransaction.tsx` file / route & related files
* Updates `useFeeCustomizer` hook to also handle EVM fees
* Updates approval screen components to the new `displayData` schema so they can be used on `GenericApprovalScreen`

## Testing
Basically a full regression on EVM features is needed:
* Send
* Bridge
* Swap
    * _all of the above features but initiated by Core Web_
* Send NFT
* dApp integrations (i.e. liquidity pool on TraderJoe.xyz)
    * make sure token spend approvals work as they used to
    * make sure LBT (Liquidity Book Tokens) collapse in an accordion
* Verify that the `chainId` parameter in `eth_sendTransaction` payload works as it used to
    * no cross-environment transaction execution allowed (no mainnet execution while UI is on testnet or vice-versa)
    * transactions on custom networks are only allowed if the custom network is also active in the UI

## Checklist for the author
- [x] I've covered new/modified business logic with Jest test cases.
- [x] I've tested the changes myself before sending it to code review and QA.

### CODE REVIEW

File: src/background/services/actions/handlers/updateTxData.ts
Comment (gergelylovas) [inline src/background/services/actions/handlers/updateTxData.ts:1]: Please add tests

File: src/background/services/network/NetworkService.ts [hunk 1]
```diff
@@ -466,7 +466,9 @@ export class NetworkService implements OnLock, OnStorageReady {
   ): Promise<NetworkWithCaipId | undefined> {
     const chainId =
       typeof scopeOrChainId === 'string'
-        ? caipToChainId(scopeOrChainId)
+        ? scopeOrChainId.startsWith('0x')
```
Comment (gergelylovas) [inline src/background/services/network/NetworkService.ts:469]: Please add a test for the hex number case

File: src/background/services/actions/ActionsService.ts [hunk 2]
```diff
@@ -212,6 +213,32 @@ export class ActionsService implements OnStorageReady {
     }
   }
 
+  async updateTx(
```
Comment (gergelylovas) [inline src/background/services/actions/ActionsService.ts:216]: Please add tests for the new method

File: src/pages/SignTransaction/components/TransactionTokenCard.tsx [hunk 3]
```diff
@@ -6,34 +6,42 @@ import {
...
 import { useEffect, useRef, useState } from 'react';
-import { useTranslation } from 'react-i18next';
 
 export enum TransactionTokenCardVariant {
   SEND = 'SEND',
   RECEIVE = 'RECEIVE',
   DEFAULT = 'DEFAULT',
 }
 
+const isNftToken = (
```
Comment (gergelylovas) [inline src/pages/SignTransaction/components/TransactionTokenCard.tsx:28]: We already have a util for this in `src/background/services/balances/nft/utils/isNFT.ts`

**Review by gergelylovas — CHANGES_REQUESTED** [review]
Nice job nice to see the update tx flow in action! Only a few small comments.

File: src/utils/calculateGasAndFees.ts [hunk 4]
```diff
@@ -39,17 +39,20 @@ export function calculateGasAndFees({
...
+    tokenDecimals
+  );
   const tip = bnTip
-    ? bigToLocaleString(bigintToBig(bnTip, tokenDecimals), 8)
+    ? bigToLocaleString(bigintToBig(bnTip, tokenDecimals), tokenDecimals)
     : null;
 
   const feeUSD = tokenPrice
-    ? parseFloat((parseFloat(fee) * tokenPrice).toFixed(4))
+    ? parseFloat((parseFloat(fee.replace(/,/g, '')) * tokenPrice).toFixed(4))
```
[the remark this answers is not in the document]
Comment (meeh0w) [reply src/utils/calculateGasAndFees.ts]: I rewritten it to use the `TokenUnit` util. b24876a09

File: src/pages/SignTransaction/components/TransactionTokenCard.tsx (hunk 3 above)
Comment (meeh0w) [reply src/pages/SignTransaction/components/TransactionTokenCard.tsx:28]: This is a different model (not a variation of `TokenWithBalance`)

File: src/background/services/network/NetworkService.ts (hunk 1 above)
Comment (meeh0w) [reply src/background/services/network/NetworkService.ts:469]: Added tests for entire `.getNetwork()` in d2632f72b

File: src/background/services/actions/ActionsService.ts (hunk 2 above)
Comment (meeh0w) [reply src/background/services/actions/ActionsService.ts:216]: Done in f011f7803

**Review by gergelylovas — APPROVED** [review]

**Review by vvava — APPROVED** [review]
🧜‍♀️

**Pull request merged**